src/pal/src/exception/signal.cpp

```diff
@@ -122,6 +122,10 @@ struct sigaction g_previous_activation;
 // Offset of the local variable containing pointer to windows style context in the common_signal_handler function.
 // This offset is relative to the frame pointer.
 int g_common_signal_handler_context_locvar_offset = 0;
+
+// Use should use initial-exec TLS model for this variable because access to it should be async-signal-safe as we
+// want to use it in common_signal_handler
+static __thread bool g_has_alternate_stack __attribute__ ((tls_model("initial-exec"))) = false;
```

Comment:
I was thinking about this a bit and I've figured we actually don't need to a add a new variable. We can use the result of the GetCurrentPalThread() - if it returns null, the current thread is foreign and so it doesn't have our alternate stack. If it returns non-null, it has the alternate stack.

Reply:
Thank you! I've removed unnecessary variable.